catshadow/client_docker_test.go

```diff
@@ -702,9 +704,22 @@ loop3:
 		}
 	}
 
+	// wait for message to be received by b
+loop3a:
```

Comment:
This fixes the symptom (test failing), but not the cause (temporal ordering / concurrency issues in `client.go`).
I suggest we make this change, and then add another RenameContact test that doesn't force the serialization order.
That is, we can't guarantee that the way this change does it is how it plays out in real life.

Reply:
I'm pretty sure the test failure is because of a bug in the test framework and there isn't a fix to client.go that would stop this problem, and I think the message storage / contact name string should not be used as a key for any of these datastructures... 

catshadow message storage needs to be redesigned; also it serializes the entire message history and writes to disk on every state change.